ui/src/dashboards/components/variablesControlBar/VariableDropdown.tsx

```diff
@@ -0,0 +1,97 @@
+// Libraries
+import React, {PureComponent} from 'react'
+import {connect} from 'react-redux'
+import _ from 'lodash'
+
+// Components
+import {Dropdown, DropdownMenuColors} from 'src/clockface'
+
+// Utils
+import {
+  getValuesForVariable,
+  getSelectedValueForVariable,
+} from 'src/variables/selectors'
+
+// Styles
+import 'src/dashboards/components/variablesControlBar/VariableDropdown.scss'
+
+// Types
+import {AppState} from 'src/types/v2'
+
+interface StateProps {
+  values: string[]
+  selectedValue: string
+}
+
+interface OwnProps {
+  name: string
```

Comment:
This and `onSelect` could come from Redux too (but it's also fine like this)

Reply:
I thought of that but I thought a utility was a lot for something we already have peeled off in the parent

Reply:
there's already a selector that could be useful: 

https://github.com/influxdata/influxdb/blob/243f1ea5112058b74f838d0eae8da8d03f4795b7/ui/src/variables/selectors/index.tsx#L115-L117

then `mstp` could look something like
```
const mstp = (state: AppState, props: OwnProps): StateProps => {
  const {variableID} = props

  const {name} = getVariable(state, variableID)

  return {name}
}
```

just as an example for how selectors can be reused.... but this is totally down the weeds and nitpicky, just wanted to share. it's totally cool as is